src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -401,6 +405,79 @@ protected override void Dispose(bool disposing)
         base.Dispose(disposing);
     }
 
+    private static Sampler ResolveSamplerFromSpecificationConfigurationKeys(IConfiguration configuration, Sampler? stateSampler)
```

Comment:
This method should also be renamed as it's not limited to just configuration. It could even return the state sampler as well. Maybe something simple like `GetSampler`.